drivers/gnss/quectel_lcx6g.c

```diff
@@ -0,0 +1,757 @@
+	uint8_t dynamic_separators_buf[2];
+	uint8_t dynamic_request_buf[32];
+	struct modem_chat_match dynamic_match;
+	struct modem_chat_script_chat dynamic_script_chat;
+	struct modem_chat_script dynamic_script;
+
+	/* Allocation for responses from GNSS modem */
+	union {
+		uint16_t fix_rate_response;
+		gnss_systems_t enabled_systems_response;
+		enum gnss_navigation_mode navigation_mode_response;
+	};
+
+	struct k_spinlock lock;
+};
+
+#define MODEM_CHAT_SCRIPT_NO_ABORT_DEFINE(_sym, _script_chats, _callback, _timeout)             \
+	static struct modem_chat_script _sym = {                                                \
+		.name = #_sym,                                                                  \
+		.script_chats = _script_chats,                                                  \
+		.script_chats_size = ARRAY_SIZE(_script_chats),                                 \
+		.abort_matches = NULL,                                                          \
+		.abort_matches_size = 0,                                                        \
+		.callback = _callback,                                                          \
+		.timeout = _timeout,                                                            \
+	}
+
+#ifdef CONFIG_PM_DEVICE
+MODEM_CHAT_MATCH_DEFINE(pair003_success_match, "$PAIR001,003,0*38", "", NULL);
+MODEM_CHAT_SCRIPT_CMDS_DEFINE(
+	suspend_script_cmds,
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR003*39", pair003_success_match)
+);
+
+MODEM_CHAT_SCRIPT_NO_ABORT_DEFINE(suspend_script, suspend_script_cmds,
+				  NULL, QUECTEL_LCX6G_SCRIPT_TIMEOUT_S);
+#endif /* CONFIG_PM_DEVICE */
+
+MODEM_CHAT_MATCH_DEFINE(any_match, "", "", NULL);
+MODEM_CHAT_MATCH_DEFINE(pair062_ack_match, "$PAIR001,062,0*3F", "", NULL);
+MODEM_CHAT_SCRIPT_CMDS_DEFINE(
+	resume_script_cmds,
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR002*38", any_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,0,1*3F", pair062_ack_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,1,0*3F", pair062_ack_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,2,0*3C", pair062_ack_match),
+#if CONFIG_GNSS_SATELLITES
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,3,5*38", pair062_ack_match),
+#else
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,3,0*3D", pair062_ack_match),
+#endif
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,4,1*3B", pair062_ack_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,5,0*3B", pair062_ack_match),
+);
+
+MODEM_CHAT_SCRIPT_NO_ABORT_DEFINE(resume_script, resume_script_cmds,
+				  NULL, QUECTEL_LCX6G_SCRIPT_TIMEOUT_S);
+
+MODEM_CHAT_MATCHES_DEFINE(unsol_matches,
+	MODEM_CHAT_MATCH_WILDCARD("$??GGA,", ",*", gnss_nmea0183_match_gga_callback),
+	MODEM_CHAT_MATCH_WILDCARD("$??RMC,", ",*", gnss_nmea0183_match_rmc_callback),
+#if CONFIG_GNSS_SATELLITES
+	MODEM_CHAT_MATCH_WILDCARD("$??GSV,", ",*", gnss_nmea0183_match_gsv_callback),
+#endif
+);
+
+static int quectel_lcx6g_configure_pps(const struct device *dev)
+{
+	struct quectel_lcx6g_config *config = (struct quectel_lcx6g_config *)dev->config;
```

Comment:
drop these castings, they buy you nothing in therm of type checking (config and data are `void *`) and in cases like this one they also hide the fact that you are losing the `const` qualifier.

Reply:
dropped